File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/main/java/simulacionBandaMusica/factories/MultiFactory.java
package simulacionBandaMusica.factories;

import simulacionBandaMusica.enums.TipoPercusion;
import simulacionBandaMusica.models.MultiInstrumentista;

import static simulacionBandaMusica.utils.sorteoDatos.*;

public class MultiFactory {

    public static MultiInstrumentista create() {
        String name = randomName();
        int experienceYear = randomAnno();
        int numCuerdas = randomNumCuerdas();
        int numTeclas = randomNumTeclas();
        TipoPercusion tipoPercusion = randomTipoPercusion();

        return new MultiInstrumentista(name, "Bajo,teclado y percusión", experienceYear, numCuerdas, numTeclas, tipoPercusion);
    }

}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/main/kotlin/simulacionBandaMusica/factories/MultiFactory.kt
package simulacionBandaMusica.factories

import simulacionBandaMusica.models.Musico
import simulacionBandaMusica.utils.sorteoDatos

object MultiFactory {
    fun create(): Musico.MultiInstrumentista {
        val name = sorteoDatos.randomName()
        val experienceYear = sorteoDatos.randomAnno()
        val numCuerdas = sorteoDatos.randomNumCuerdas()
        val numTeclas = sorteoDatos.randomNumTeclas()
        val tipoPercusion = sorteoDatos.randomTipoPercusion()
        return Musico.MultiInstrumentista(
            name,
            "Bajo,teclado y percusión",
            experienceYear,
            numCuerdas,
            numTeclas,
            tipoPercusion
        )
    }
}